### PR TITLE
Update template to the latest version of the USWDS-Jekyll 

### PR DESCRIPTION
**Pull request by line47**
Preview: https://federalist-proxy.app.cloud.gov/preview/18f/federalist-uswds-template/update-uswds/docs/

This update includes
- [Jekyll pages API search](https://github.com/18F/jekyll_pages_api_search) is now available 
- USWDS version 1.4.3
- modification of page layout names to work with `Jekyll new` commands

More info here: https://github.com/18F/uswds-jekyll

### CODE REVIEW

**Review by wslack — CHANGES_REQUESTED** [review]
The "section one" and "section two" links in the documentation are broken by this change. Possible not to include that behavior?

**Comment by wslack** [comment]
Compare: https://federalist-proxy.app.cloud.gov/preview/18f/federalist-uswds-template/update-uswds/docs/

and https://federalist-uswds-template.18f.gov/docs/#section-one

**Comment by line47** [comment]
This has been fixed, if you could take another look @wslack

**Comment by wslack** [comment]
Is the screen supposed to flash when clicking on "Site Title" from https://federalist-proxy.app.cloud.gov/preview/18f/federalist-uswds-template/update-uswds/docs/ ?

**Comment by line47** [comment]
It looks like the Jekyll_api_search is what is causing the issue. 😞 

When I comment out that code I no longer get the flashing type of page load. 
I'm going to remove the search for now as the current version does not have search either.

**Review by wslack — APPROVED** [review]
Looks like feature parity to me!

**Comment by line47** [comment]
The `usa-overlay` issue has been fixed (via a workaround) 🎉  ready for a second review @wslack

File: _data/header.yml [hunk 1]
```diff
@@ -11,12 +11,11 @@ primary:
 
 secondary:
   # this is a key into _data/navigation.yml
-  links: secondary
+  links:
 
   # to enable search, change this to an object with 'action' and
```
Comment (jseppi) [inline _data/header.yml]: Comment seems out of date now that `search` is enabled by default

Comment (jseppi) [reply _data/header.yml]: Actually I wonder if we _should_ keep this enabled by default. This plugin seems to generate odd results. For instance, I searched for ["content" at the preview build](https://federalist-proxy.app.cloud.gov/preview/18f/federalist-uswds-template/update-uswds/search/?q=content), and here are the results:

screen shot 2017-12-15 at 11 47 13 am

That blank entry for number 2 is odd, but it is something I notice happening with a lot of sites that use the `jekyll_pages_api_search` plugin.

Comment (line47) [reply _data/header.yml]: I'm totally good with leaving search out in favor of moving to search.gov in a future update

**Comment by line47** [comment]
Removed the Jekyll search want to take a final look @wslack @jseppi  ?
😎  https://federalist-proxy.app.cloud.gov/preview/18f/federalist-uswds-template/update-uswds/

File: Gemfile [hunk 2]
```diff
@@ -1,3 +1,7 @@
 source "https://rubygems.org"
 
 gem 'uswds-jekyll', :git => 'https://github.com/18F/uswds-jekyll.git'
+
+group :jekyll_plugins do
+  gem 'jekyll_pages_api_search'
```
Comment (jseppi) [inline Gemfile:6]: should this be removed also (and in the `Gemfile.lock`)?

Comment (line47) [reply Gemfile:6]: The `Jekyll_pages_api_search` is part of the USWDS-Jekyll theme and removing the gem from the `gemfile` and `gemfile.lock` throws errors. So maybe we can leave it in for now and remove once we remove `Jekyll_pages_api_search` from USWDS-Jekyll.

Comment (jseppi) [reply Gemfile:6]: oof, this is bringing back bad memories :P

**Review by jseppi — APPROVED** [review]

File: Gemfile (hunk 2 above)
Comment (line47) [reply Gemfile:6]: I've got an issue to incorporate search.gov as a search option https://github.com/18F/uswds-jekyll/issues/62

Comment (wslack) [reply Gemfile:6]: It's fine to do that in another PR.

**Review by wslack — APPROVED** [review]

**Pull request merged**